test/libsolidity/ASTJSON/assembly/nested_functions.json

```diff
@@ -130,8 +130,8 @@
                 "storageLocation": "default",
                 "typeDescriptions":
                 {
-                  "typeIdentifier": "t_uint256",
-                  "typeString": "uint256"
+                  "typeIdentifier": null,
+                  "typeString": null
```

Comment:
Hm - this is something to think about... those are AST tests for sources that fail the analysis and now the analysis stops before all types are resolved.
How much do we care about accuracy of ASTs in cases that fail the analysis?
To handle all this properly, we would probably have to re-evaluate the fatal-ness of errors and the stages for checks... I'll also have another look at tweaking the order of things in the compiler stack...

Reply:
Put differently: should we make typing of declarations robust against failures to resolve declarations and still assign types whereever we can?

Reply:
As a first thought, I think if analysis fails the AST isn't useful, but then there's the error recovery case for which it might be useful

Reply:
Yeah... although in the error recovery case, it's fine if the AST is "as good as possible"... in this particular case it could make sense to try to make it a bit more graceful, though...

Reply:
I think our prime concern is the correctness of the compiler. Making it useful on invalid source code is a secondary objective and I actually think that on invalid source, a parser-generator based system that only performs very crucial checks might be the better solution.